### PR TITLE
Rescue of CGMES models

### PR DESCRIPTION
**Pull request by dietmarw**
I still got this old branch from Francis which was simply named "CGMES".  In order to not lose the work I now rebased it on current master and fixed it up so that the models are at least accessible and even simulate. Now I'm not sure if these models are still valuable and of interest to get in proper shape and merged into the library: 

From the commit the list of added models seem to be:
 - `Electrical.Controls.CGMES.TG.GovHydroIEEE0` with `Examples.Controls.CGMES.TG.GovHydroIEEE0_Test`
 - `Examples.Machines.PSSE.Plant` package which seems to test some new layout.

Maybe some of you remember the background and can decide as to what to do with those models.

### CODE REVIEW

**Comment by marcelofcastro** [comment]
I reviewed the files and they look fine. However, I will wait for @lvanfretti to take a look at what you said @dietmarw. Maybe he has an idea of what he wants to do with all the models. I will approve the merge after we discuss this :)

**Comment by dietmarw** [comment]
Yes I don't want them merged as they are right now anyway. I first want to know if it makes sense to invest some more clean up and documentation work.

**Comment by lvanfretti** [comment]
@dietmarw these models were developed for the ENTSO-E interoperability testing in 2016, see this [link](https://eepublicdownloads.entsoe.eu/clean-documents/CIM_documents/IOP/160715_CGMES_IOPreport2016.pdf).

The goal of the models was to have the most basic dynamic components specified by the standard that could be mapped to OpenIPSL. The vendors have a similar small model to compare their results, so this was going to be used as the "open source reference".

Since I left in 2017, I did not follow up on the ENTSO-E IOP efforts.

So, unless you want to keep them, we really have not motivation to do so. If you want to keep them, a good starting point would be to check if the control models used are not just a copy of the ones under the PSS/E models, which I think might be the case.

Is this the kind of information you were looking for?

**Comment by dietmarw** [comment]
All I need is to know if there is still value in the models added in this PR wrt the goal/vision of OpenIPSL (whatever those are ;-) ). Would be a shame to throw something away that has been worked on and can still be of value. If not that's fine with me too.

**Comment by lvanfretti** [comment]
@dietmarw then if they are working we can keep them.

**Comment by dietmarw** [comment]
OK next question what model is `GovHydroIEEE0` exactly. Any documentation reference? If not then even functioning models are not much of use.

I've figured out what the Plant model is about. So that one is probably not important to keep.

**Comment by lvanfretti** [comment]
@dietmarw I've sent you some information on the Keybase chat.

File: OpenIPSL/Electrical/Controls/CGMES/TG/GovHydroIEEE0.mo
Comment (marcelofcastro) [inline OpenIPSL/Electrical/Controls/CGMES/TG/GovHydroIEEE0.mo:29]: it might be -T4 here.

Comment (marcelofcastro) [inline OpenIPSL/Electrical/Controls/CGMES/TG/GovHydroIEEE0.mo:29]:
```suggestion
    b={-T4,1},
```

**Comment by dietmarw** [comment]
I think we can narrow down the contribution of this branch to adding: 
  - `Electrical.Controls.CGMES.ES.ExcSEXS`
 - `Electrical.Controls.CGMES.TG.GovHydroIEEE0`

So the question remain: 
 1. Is `ExcSEXS` useful in addition to the existing `SEXS`
 2. What IEEE reference is `GovHydroIEEE0` actually based on.

When either of both should be included we really should have the backing documentation/references in place to point to. Which then also explains the time constants meaning etc.

**Comment by lvanfretti** [comment]
Hi @dietmarw 

Ideally we should follow the detailed description of ExcSEXS and GovHydroIEEE0 that are in the IEC 61970-302 standard. Unfortunately, this is a "closed access" standard and we can only see the table of contents in the following [link](https://webstore.iec.ch/preview/info_iec61970-302%7Bed1.0%7Db.pdf). 
Don't get me started about how bad it is that every standard is behind a paywall... I don't like it either.

I have uploaded a few other documents to the Keybase that can help us. The very large .pdf TC57CIM... was generated from the "Enterprise Architect" tool that was used to define the UML of the dynamics profile. Page 642-643 have the ExcSEXS info, while GovHydroIEEE0 is in pages 309-311.

There is no literature reference to ExcSEXS, but this is almost the same as the normal SEXS, except for the naming. So, the IEEE standard Marcelo mentioned earlier should be fine for that.

For GovHydroIEEE0 it says that the reference is this one: https://ieeexplore.ieee.org/document/4075292
However, in page 288, it is stated that: "A 2012 IEEE report, Dynamic Models for Turbine-Governors in Power System Studies, provides updated information on a variety of models including IEEE, vendor and reliability authority models. Fully incorporating the results of that report into the CIM Dynamics model is a future effort." That report is here: https://site.ieee.org/fw-pes/files/2013/01/PES_TR1.pdf

Hope this helps!

**Comment by lvanfretti** [comment]
@dietmarw I have now compared the difference between the documentation of two tools and the CGMES documentation I have. I have sent you a summary in the Keybase with my recommendation.
@marcelofcastro this is a simple model, if it doesn't match the CGMES documentation, then you can help @dietmarw implement it if he doesn't have the time.

**Comment by dietmarw** [comment]
@lvanfretti  I've had a look at the [https://site.ieee.org/fw-pes/files/2013/01/PES_TR1.pdf](PES_TR1.pdf) document and it says: on page 4-1:

> This simple representation can only provide accuracy within a very limited
bandwidth, and is therefore inadequate for general use. For the most part, use of
this model has disappeared in large system databases, but vestiges of these models
for older plants are still occasionally found. This model, sometimes referred to as
the IEEE Type 2 model is obsolete, and inclusion of this model structure in
commercial programs should be discontinued.

So this makes me wonder how much sense makes to include the apparently outdated model `GovHydroIEEE0` at all. Or are there any verification examples that still depend on this?

**Comment by lvanfretti** [comment]
@dietmarw the document is only referring to be it discontinued in "commercial" programs, and that's really narrowly setting the stage to the ones use by the IEEE people in that group (i.e. the PSS/E and PSLF models...)

If it is included in the CGMES, it means that it is not discontinued in Europe, in fact, then it is part of the standard. Moreover, they state that "vestiges" are still found in some places. That is probably an under-statement only referring to the models discussed by the people involved in that working group.

My suggestion is to put a note on the model, where we would recommend to use an alternative instead of it, and refer to the IEEE document. After all, if the user only has the data to parametrize that model and wants to compare it to an alternative, we should keep the models so they can verify for themselves the recommendations in the IEEE document.

**Comment by dietmarw** [comment]
@lvanfretti OK in that case I've got the two models ready. Just waiting for #249 to get merged so I can rebase my fixes to this PR onto master.

**Comment by lvanfretti** [comment]
@dietmarw that sounds great! The new version is coming up to be a great improvement thanks to you and @marcelofcastro

**Comment by dietmarw** [comment]
Ready for review.

**Review by marcelofcastro — APPROVED** [review]
Merging is approved.

**Pull request merged**
